### ISSUE TITLE
Allow specifying pin file location when running @repo//:pin

### ISSUE THREAD

**Issue by kamalmarhubi** [bazelbuild/rules_jvm_external#436]
We have multiple disparate environments (Android, YARN, "plain" jvm) that have disjoint dependencies, and we'd like to separate them all out. It's easy to do, but slightly clunky, as you have to pin and then move the file to be closer to the code and avoid cluttering up the repo root.

The pinning actually works fine if you move the file, and will correctly update the new file when you change the dependencies. The first run, however, puts the file at the repository root.

**Comment by jin** [comment]
Yeah, the intention is to store the file at the root of the repo on the initial pin to keep things simple. Most folks would keep the file at that location. I don't think it's worth adding complexity to the pin script to make this configurable, since further updates to a moved maven_install.json does work.
